jgrapht-core/src/main/java/org/jgrapht/experimental/dag/DirectedAcyclicGraph.java

```diff
@@ -532,6 +533,82 @@ private void reorder(Set<V> df, Set<V> db, Visited visited)
     }
 
     /**
+     * This function is implemented recursively. Therefore only graphs without
+     * any cycles are allowed.
+     *
+     * @param graph graph to look for ancestors in.
+     * @param vertex the vertex to get the ancestors of.
+     *
+     * @return {@link List} of ancestors of the vertex in the given graph.
+     */
+    public static <V, E extends DefaultEdge> List<V> getAncestors(
+        DirectedAcyclicGraph<V, E> graph,
+        V vertex)
+    {
+
+        List<V> ancestors = new ArrayList<>();
+        ancestors = getAncestors(graph, vertex, ancestors);
+        return ancestors;
+    }
+
+    protected static <V, E extends DefaultEdge> List<V> getAncestors(
```

Comment:
Also, to avoid blowing the stack on very deep graphs, you could use DepthFirstIterator instead.